### PR TITLE
mime type fix

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
mime type fix for lighthouse claim doc upload. 

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/4746

## Testing done
staging and spec

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/lighthouse/benefits_documents/configuration.rb [hunk 1]
```diff
@@ -93,7 +93,8 @@ def generate_upload_body(document_data, file_body)
       file = Tempfile.new(document_data[:file_name])
       File.write(file, file_body)
 
-      payload[:file] = Faraday::UploadIO.new(file, file_type)
+      file_mime_type = MimeMagic.by_path(document_data[:file_name]).type
+      payload[:file] = Faraday::UploadIO.new(file, file_mime_type)
```
Comment (aherzberg) [inline lib/lighthouse/benefits_documents/configuration.rb:97]: mixed up the `subtype` and `type`. subtype is the shorthand (jpg) while type is the actual mime type that the file needs (image/jpg).

File: spec/lib/lighthouse/benefits_documents/configuration_spec.rb [hunk 2]
```diff
@@ -40,10 +40,10 @@
       let(:file_body) { File.read(fixture_file_upload('doctors-note.jpg', 'image/jpeg')) }
 
       it 'uploads expected document' do
-        VCR.use_cassette('lighthouse/benefits_claims/documents/lighthouse_document_upload_200') do
+        VCR.use_cassette('lighthouse/benefits_claims/documents/lighthouse_document_upload_200_jpg') do
```
Comment (aherzberg) [inline spec/lib/lighthouse/benefits_documents/configuration_spec.rb:43]: updated jpg spec to use real staging response.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1832,6 +1832,8 @@ spec/support/vcr_cassettes/lighthouse/auth/client_credentials/invalid_client_id_
 spec/support/vcr_cassettes/lighthouse/auth/client_credentials/invalid_scopes_400.yml @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group @department-of-veterans-affairs/vfs-facilities
 spec/support/vcr_cassettes/lighthouse/auth/client_credentials/revoke_consent_204.yml @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group @department-of-veterans-affairs/vfs-facilities
 spec/support/vcr_cassettes/lighthouse/auth/client_credentials/token_200.yml @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group @department-of-veterans-affairs/vfs-facilities
+spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_document_upload_200_jpg.yml @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
+spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_document_upload_200_pdf.yml @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
```
Comment (aherzberg) [inline .github/CODEOWNERS:1836]: will this be required for all new files added to vets-api? seems like there should be some higher level rules rather then file by file

**Review by kpethtel — APPROVED** [review]
Looks good AFAICT. Can't say anything about the codeowners part.

File: .github/CODEOWNERS (hunk 3 above)
Comment (rmtolmach) [reply .github/CODEOWNERS:1836]: You can add one line for `spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/`, or even `spec/support/vcr_cassettes/lighthouse/benefits_claims/` if all files in that dir would have the same owners.

**Review by Tonksthebear — DISMISSED** [review]
LGTM

File: .github/CODEOWNERS (hunk 3 above)
Comment (aherzberg) [reply .github/CODEOWNERS]: oh, good to know. thanks, updated to cover the directory.

**Review by mathisto — APPROVED** [review]

**Review by kpethtel — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]
codeowners changes look good

**Pull request merged**